dali/ft/filecopy.cpp

```diff
@@ -465,29 +465,34 @@ FileSizeThread::FileSizeThread(FilePartInfoArray & _queue, CriticalSection & _cs
 
 bool FileSizeThread::wait(unsigned timems)
 {
-    while (!sem.wait(timems))   { // report every time
-        if (!cur.get())
-            continue;       // window here?
-        cur->Link();
-        RemoteFilename *rfn=NULL;
-        if (copy) {
-            if (!cur->mirrorFilename.isNull())
-                rfn = &cur->mirrorFilename;
-        }
-        else {
-            rfn = &cur->filename;
-        }
-        if (rfn) {
+    while (!sem.wait(timems))
+    { // report every time
+        StringBuffer rfn;
+        {
+            CriticalBlock lock(cs);
             StringBuffer url;
-            WARNLOG("Waiting for file: %s",rfn->getRemotePath(url).str());
-            cur->Release();
+            if (cur.get())
+            {
+                if (copy)
+                {
+                    if (!cur->mirrorFilename.isNull())
+                        rfn.set(cur->mirrorFilename.getRemotePath(url));
```

Comment:
you don't need both 'rfn' and 'utl' StringBuffer's.
You can just pass 'rfn' to getRemoteBuffer only, i.e.:
```
   cur->mirrorFilename.getRemotePath(rn));
```

and remove line 473

Reply:
True, changed.